encoder/avro.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * skogul, test avro encoder
+ *
+ * Copyright (c) 2022 Telenor Norge AS
+ * Author:
+ *  - Roshini Narasimha Raghavan <roshiragavi@gmail.com>
+  *  - Kristian Lyngstøl <kly@kly.no>
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+*/
+
+package encoder
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/hamba/avro"
+	"github.com/telenornms/skogul"
+)
+
+type AVRO struct {
+	Schema string
+}
+
+func (x AVRO) Encode(c *skogul.Container) ([]byte, error) {
+	b, _ := os.ReadFile(x.Schema)
+	schema := avro.MustParse(string(b))
```

Comment:
This part needs to be inside sync.Once, similar to how it's done in transformer/timestamp.go for instance. Otherwise Skogul will open the schema on disk for every single AVRO packet, which could be several thousand per second.

It ALSO needs to check the return code of os.ReadFile(), so change `b, _ :=` to `b, err := ` and check if err is nil or not.